107/src/main/java/org/ehcache/jsr107/Eh107CacheStatisticsMXBean.java

```diff
@@ -15,179 +15,211 @@
  */
 package org.ehcache.jsr107;
 
-import java.util.EnumMap;
-import java.util.Map.Entry;
-import java.util.concurrent.ConcurrentMap;
+import org.ehcache.Cache;
+import org.ehcache.Ehcache;
+import org.ehcache.EhcacheHackAccessor;
+import org.ehcache.statistics.BulkOps;
+import org.ehcache.statistics.CacheOperationOutcomes;
+import org.terracotta.context.TreeNode;
+import org.terracotta.context.query.Matcher;
+import org.terracotta.context.query.Matchers;
+import org.terracotta.context.query.Query;
+import org.terracotta.statistics.OperationStatistic;
+import org.terracotta.statistics.StatisticsManager;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.Set;
 import java.util.concurrent.atomic.AtomicLong;
 
-import javax.cache.management.CacheStatisticsMXBean;
-
-import org.ehcache.statistics.BulkOps;
-import org.ehcache.statistics.CacheStatistics;
+import static org.terracotta.context.query.Matchers.attributes;
+import static org.terracotta.context.query.Matchers.context;
+import static org.terracotta.context.query.Matchers.hasAttribute;
+import static org.terracotta.context.query.QueryBuilder.queryBuilder;
 
 /**
- * @author teck
+ * @author Ludovic Orban
  */
-class Eh107CacheStatisticsMXBean extends Eh107MXBean implements CacheStatisticsMXBean {
-
-  private final org.ehcache.statistics.CacheStatistics statistics;
-  private final ConcurrentMap<BulkOps, AtomicLong> bulkMethodCounts;
-  private final EnumMap<Stat, AtomicLong> baselines = new EnumMap<Stat, AtomicLong>(Stat.class);
-
-  Eh107CacheStatisticsMXBean(String cacheName, Eh107CacheManager cacheManager,
-      org.ehcache.statistics.CacheStatistics statistics) {
+public class Eh107CacheStatisticsMXBean extends Eh107MXBean implements javax.cache.management.CacheStatisticsMXBean {
+
+  private final Ehcache cache;
```

Comment:
From looking at the code, we could get a hold of the `bulkMethodEntries` in the constructor and thus not hold a reference to the cache.